src/syrupy/extensions/single_file.py

```diff
@@ -69,5 +69,12 @@ def _write_snapshot_fossil(self, *, snapshot_fossil: "SnapshotFossil") -> None:
 
     def __clean_filename(self, filename: str) -> str:
         filename = str(filename).strip().replace(" ", "_")
-        max_filename_length = 255 - len(self._file_extension or "")
-        return re.sub(r"(?u)[^-\w.]", "", filename)[:max_filename_length]
+        exclude_chars = '\\/:*"<>|?'
+        exclude_categ = ("C",)
+        cleaned_filename = "".join(
+            c
+            for c in filename
+            if c not in exclude_chars
+            and not any(categ in category(c) for categ in exclude_categ)
+        )
+        return cleaned_filename[:255]  # maximum filename length
```

Comment:
Why remove the `- len(self._file_extension)`?

Reply:
Oops reverted